[Bug] MultiFragmentTest.partitionedOutputWithLargeInput hangs

### Component Selection

- [ ] Core Engine (Expression eval, Memory, Vector)
- [ ] Connectors / File Formats (Hive, Parquet, etc.)
- [ ] API / Bindings (Python, etc.)
- [x] Build
- [ ] Other

### Describe the Bug

Occasionally when running tests in CI, we see the build pipeline stalls when running tests. Usually test runs take ~10 minutes, but sometimes it will hang for an hour and then fail. This consumes CI resources and delays valuable PR feedback.

The culprit seems to be `MultiFragmentTest.partitionedOutputWithLargeInput`. A sample fragment from the output log of a CI failure is reproduced below:

```
[       OK ] MultiFragmentTest.mergeExchange (2277 ms)
[ RUN      ] MultiFragmentTest.partitionedOutput
[       OK ] MultiFragmentTest.partitionedOutput (1620 ms)
[ RUN      ] MultiFragmentTest.partitionedOutputWithLargeInput
E20260302 07:06:12.811446 126393690085056 Exceptions.h:82] Line: /__w/bolt/bolt/bolt/common/memory/MemoryArbitrator.cpp:131, Function:growCapacity, Expression:  Exceeded memory pool capacity. Exceeded memory pool capacity.Can't grow default_root_239 capacity with 1.00MB. This will exceed its max capacity 1.00MB, current capacity 1.00MB.
Memory Pool[default_root_239 AGGREGATE root[default_root_239] parent[null] MALLOC track-usage thread-safe]<max capacity 1.00MB capacity 1.00MB used 980.00KB available 0B reservation [used 0B, reserved 1.00MB, min 0B] counters [allocs 0, frees 0, reserves 0, releases 0, collisions 0])>
default_root_239 usage 980.00KB reserved 1.00MB peak 1.00MB
    task.local://leaf-0 usage 980.00KB reserved 1.00MB peak 1.00MB
        node.1 usage 980.00KB reserved 1.00MB peak 1.00MB
            op.1.0.0.PartitionedOutput usage 980.00KB reserved 1.00MB peak 980.00KB

Top 1 leaf memory pool usages:
    op.1.0.0.PartitionedOutput usage 980.00KB reserved 1.00MB peak 980.00KB
, Source: RUNTIME, ErrorCode: MEM_CAP_EXCEEDED


99% tests passed, 1 tests failed out of 408

Label Time Summary:
celeborn_e2e    =   0.01 sec*proc (1 test)

Total Test time (real) = 3044.95 sec

The following tests did not run:
	 87 - */MemoryAllocatorTest.allocContiguousVsize/* (Disabled)
	125 - */MemoryCapExceededTest.singleDriver/* (Disabled)
	127 - */MemoryCapExceededTest.allocatorCapacityExceededError/* (Disabled)
	148 - */MemoryPoolTest.memoryLeakCheck/* (Disabled)
	187 - */MemoryPoolTest.concurrentUpdateToSharedPools/* (Disabled)
	279 - */SharedArbitrationTestWithThreadingModes.raceBetweenTaskTerminateAndReclaim/* (Disabled)

The following tests FAILED:
	401 - bolt_exec_test (Timeout)
make: *** [Makefile:362: unittest_release_spark] Error 8
Error: Process completed with exit code 2.

```

This shows that `MultiFragmentTest.partitionedOutputWithLargeInput` started, but never finished. There is one output log message about `MEM_CAP_EXCEEDED` but doesn't describe why the test hangs.



### Reproduction Steps

1. Create a PR
2. Run the CI 5-10 times
3. Observe runtime. It is likely to have at least one hang

### Bolt Version / Commit ID

4a3764b694071db6d82cfbb53835769a019bbe6b

### System Configuration

```markdown
CI machines
```

### Logs / Stack Trace

```shell
See above
```

### Expected Behavior

No hang. The test should either fail immediately, or pass in a reasonable time window.

### Additional context

_No response_

--- a/bolt/exec/tests/MultiFragmentTest.cpp
+++ b/bolt/exec/tests/MultiFragmentTest.cpp
@@ -602,8 +602,7 @@ TEST_F(MultiFragmentTest, partitionedOutputWithLargeInput) {
     leafTask->start(1);
     auto op = PlanBuilder().exchange(leafPlan->outputType()).planNode();
 
-    auto task =
-        assertQuery(op, {leafTaskId}, "SELECT c0, c1, c2, c3, c4 FROM tmp");
+    assertQuery(op, {leafTaskId}, "SELECT c0, c1, c2, c3, c4 FROM tmp");
     ASSERT_TRUE(waitForTaskCompletion(leafTask.get()))
         << leafTask->taskId() << "state: " << leafTask->state();
     // Check the leaf task's PartitionedOutput operator ("1") for outputVectors.
@@ -647,8 +646,7 @@ TEST_F(MultiFragmentTest, partitionedOutputWithLargeInput) {
 
     auto op = PlanBuilder().exchange(intermediatePlan->outputType()).planNode();
 
-    auto task = assertQuery(
-        op, intermediateTaskIds, "SELECT c0, c1, c2, c3, c4 FROM tmp");
+    assertQuery(op, intermediateTaskIds, "SELECT c0, c1, c2, c3, c4 FROM tmp");
     ASSERT_TRUE(waitForTaskCompletion(leafTask.get()))
         << "state: " << leafTask->state();
     // Same as above: check the leaf's PartitionedOutput operator ("1").

